BH TTNN Unit test failure in moreh linear/matmul

tests/ttnn/unit_tests/operations/test_moreh_linear.py::test_moreh_bias_backward_fp32 fails in CI for BH

https://github.com/tenstorrent/tt-metal/actions/runs/14041966068/job/39314377311
```
        logger.debug(f"Out passing={passing}")
        logger.debug(f"Output pcc={output_pcc}")
>       assert passing
E       assert False

tests/ttnn/unit_tests/operations/test_moreh_linear.py:498: AssertionError
...
FAILED tests/ttnn/unit_tests/operations/test_moreh_linear.py::test_moreh_bias_backward_fp32[shapes=([8, 512, 768], [2304, 768], [1, 2304], [8, 512, 2304])] - assert False
```

Also seems to affect
```
FAILED tests/ttnn/unit_tests/operations/test_moreh_matmul.py::test_moreh_matmul_fp32_dest_acc[params=([32, 3200], [3200, 32], [32, 32], False, False)] - assert False
```

Should run all moreh ops.

This is happening in another test as well:
https://github.com/tenstorrent/tt-metal/actions/runs/14143705878/job/39628534456
```
FAILED tests/ttnn/unit_tests/operations/test_moreh_matmul.py::test_moreh_matmul_fp32_dest_acc[params=([32, 3200], [3200, 32], [32, 32], False, False)] - assert False
```
Will need to add more skips.

And another test

https://github.com/tenstorrent/tt-metal/actions/runs/14279928144/job/40028289639
```
FAILED tests/ttnn/unit_tests/operations/test_moreh_matmul.py::test_moreh_matmul_with_bias_add_fp32_dest_acc[params=([31, 3100], [3100, 31], [31, 31], False, False)] - assert False
```

And another test


https://github.com/tenstorrent/tt-metal/actions/runs/14279928144/job/40028289490
```
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_shape[shape=[1, 31, 639, 31]] - assert False
```

Local pytests of the files:
```
pytest tests/ttnn/unit_tests/operations/test_moreh_matmul.py
FAILED tests/ttnn/unit_tests/operations/test_moreh_matmul.py::test_moreh_matmul_with_bias_add_fp32_dest_acc[params=([31, 3100], [3100, 31], [31, 31], False, False)]

pytest tests/ttnn/unit_tests/operations/test_moreh_bmm.py
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_shape[shape=[1, 31, 639, 31]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_shape[shape=[5, 95, 415, 65]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_shape[shape=[10, 77, 320, 320]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_shape[shape=[10, 191, 447, 159]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_optional_output[optional_output=False] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_optional_output[optional_output=True] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_ttnn_dtype[ttnn_dtype=DataType.BFLOAT16] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_callback[shape=[10, 191, 447, 159]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_backward_shape[shape=[10, 77, 320, 320]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_backward_shape[shape=[7, 511, 313, 765]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_backward_requires_grad[requires_grad=[True, False]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_backward_requires_grad[requires_grad=[False, True]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_backward_requires_grad[requires_grad=[True, True]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_backward_ttnn_dtype[ttnn_dtype=DataType.BFLOAT16] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_backward_callback[requires_grad=[True, False]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_backward_callback[requires_grad=[False, True]] - assert False
FAILED tests/ttnn/unit_tests/operations/test_moreh_bmm.py::test_moreh_bmm_backward_callback[requires_grad=[True, True]] - assert False
```

--- a/tests/ttnn/unit_tests/operations/test_moreh_bmm.py
+++ b/tests/ttnn/unit_tests/operations/test_moreh_bmm.py
@@ -8,7 +8,7 @@ import torch.nn.functional as F
 from loguru import logger
 
 import ttnn
-from models.utility_functions import comp_allclose_and_pcc, is_grayskull, skip_for_blackhole
+from models.utility_functions import comp_allclose_and_pcc, is_grayskull
 from tests.ttnn.unit_tests.operations.test_utils import (
     compute_kernel_ids,
     compute_kernel_options,
@@ -195,7 +195,6 @@ def run_moreh_bmm_backward(
         assert passing
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize(
     "shape",
     [
@@ -213,7 +212,6 @@ def test_moreh_bmm_shape(shape, device):
     run_moreh_bmm(shape, True, False if is_grayskull() else True, device)
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize("optional_output", [False, True])
 def test_moreh_bmm_optional_output(optional_output, device):
     """
@@ -232,7 +230,6 @@ def test_moreh_bmm_compute_kernel_options(compute_kernel_options, device):
     run_moreh_bmm([10, 191, 447, 159], True, compute_kernel_options, device)
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize("ttnn_dtype", [ttnn.bfloat8_b, ttnn.bfloat16])
 def test_moreh_bmm_ttnn_dtype(ttnn_dtype, device):
     """
@@ -245,7 +242,6 @@ def test_moreh_bmm_ttnn_dtype(ttnn_dtype, device):
     run_moreh_bmm([10, 191, 447, 159], True, False if is_grayskull() else True, device, ttnn_dtype=ttnn_dtype)
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize(
     "shape",
     [[10, 191, 447, 159]],
@@ -267,7 +263,6 @@ def test_moreh_bmm_callback(shape, device, use_program_cache):
     assert num_program_cache_entries_list[0] == num_program_cache_entries_list[1]
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize(
     "shape",
     [
@@ -285,7 +280,6 @@ def test_moreh_bmm_backward_shape(shape, device):
     run_moreh_bmm_backward(shape, [True, True], False if is_grayskull() else True, device)
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize(
     "requires_grad",
     [
@@ -311,7 +305,6 @@ def test_moreh_bmm_backward_compute_kernel_options(compute_kernel_options, devic
     run_moreh_bmm_backward([7, 511, 313, 765], [True, True], compute_kernel_options, device)
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize("ttnn_dtype", [ttnn.bfloat8_b, ttnn.bfloat16])
 def test_moreh_bmm_backward_ttnn_dtype(ttnn_dtype, device):
     """
@@ -326,7 +319,6 @@ def test_moreh_bmm_backward_ttnn_dtype(ttnn_dtype, device):
     )
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize(
     "requires_grad",
     [

--- a/tests/ttnn/unit_tests/operations/test_moreh_linear.py
+++ b/tests/ttnn/unit_tests/operations/test_moreh_linear.py
@@ -5,7 +5,7 @@
 import pytest
 import torch
 import ttnn
-from models.utility_functions import comp_allclose_and_pcc, skip_for_blackhole
+from models.utility_functions import comp_allclose_and_pcc
 from loguru import logger
 from tests.ttnn.unit_tests.operations.test_utils import (
     get_compute_kernel_options,
@@ -392,7 +392,6 @@ def test_moreh_linear_backward_enable_cache(shapes, device, use_program_cache):
     assert num_program_cache_entries_list[0] == num_program_cache_entries_list[1]
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize(
     "shapes",
     (

--- a/tests/ttnn/unit_tests/operations/test_moreh_matmul.py
+++ b/tests/ttnn/unit_tests/operations/test_moreh_matmul.py
@@ -8,7 +8,7 @@ from loguru import logger
 
 import ttnn
 import ttnn.operations
-from models.utility_functions import comp_allclose_and_pcc, skip_for_blackhole
+from models.utility_functions import comp_allclose_and_pcc
 from tests.ttnn.unit_tests.operations.test_utils import (
     get_compute_kernel_options,
     compute_kernel_options,
@@ -320,7 +320,6 @@ def test_moreh_matmul_enable_cache(params, device, use_program_cache):
     assert device.num_program_cache_entries() == 2
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize(
     "params",
     (
@@ -417,7 +416,6 @@ def test_moreh_matmul_backward(params, requires_grad, dtype, device):
     moreh_matmul_backward(params, requires_grad, device, dtype=dtype)
 
 
-@skip_for_blackhole("Fails on BH. Issue #19639")
 @pytest.mark.parametrize(
     "params",
     (

--- a/tests/ttnn/unit_tests/operations/test_utils.py
+++ b/tests/ttnn/unit_tests/operations/test_utils.py
@@ -8,38 +8,24 @@ import pytest
 import torch
 
 import ttnn
-from models.utility_functions import is_wormhole_b0
 
 TILE_HEIGHT = 32
 TILE_WIDTH = 32
 
 
-compute_kernel_options = [False]
-compute_kernel_ids = ["fp32_dest_acc_en=False"]
-if is_wormhole_b0():
-    compute_kernel_options.append(True)
-    compute_kernel_ids.append("fp32_dest_acc_en=True")
+compute_kernel_options = [False, True]
+compute_kernel_ids = ["fp32_dest_acc_en=False", "fp32_dest_acc_en=True"]
 
 
-def get_compute_kernel_options(compute_kernel_options):
-    if compute_kernel_options is None:
+def get_compute_kernel_options(fp32_dest_acc_en):
+    if fp32_dest_acc_en is None:
         return None
-    if is_wormhole_b0():
-        fp32_dest_acc_en = compute_kernel_options
-        packer_l1_acc = False
-        compute_kernel_config = ttnn.WormholeComputeKernelConfig(
-            math_fidelity=ttnn.MathFidelity.HiFi4,
-            math_approx_mode=False,
-            fp32_dest_acc_en=fp32_dest_acc_en,
-            packer_l1_acc=packer_l1_acc,
-        )
-    else:
-        # Grayskull doesn't support FP32, but passing a Grayskull config in the test is OK.
-        compute_kernel_config = ttnn.GrayskullComputeKernelConfig(
-            math_fidelity=ttnn.MathFidelity.HiFi4,
-            math_approx_mode=True,
-        )
-    return compute_kernel_config
+    return ttnn.WormholeComputeKernelConfig(
+        math_fidelity=ttnn.MathFidelity.HiFi4,
+        math_approx_mode=False,
+        fp32_dest_acc_en=fp32_dest_acc_en,
+        packer_l1_acc=False,
+    )
 
 
 def to_torch(ttnn_tensor, *, shape=None):

